### ISSUE TITLE
Implement stack frame inspection

### ISSUE THREAD

**Issue by wesgarland** [wesgarland/niim#17]
We need commands like `up` and `down` in gdb which allow the current operations (print, display, exec, repl) to operate on different stack frames once the debugger point has been hit.

There's some information about that here https://github.com/nodejs/node/issues/14049

@jkrems says 

> There's no official/documented way of doing it right now. Some things that work but definitely need some cleanup and an official interface:

```
debug> bt[2].list() // list source context for frame at index 2
debug> bt[2].loadScopes() // show the scope chain w/ values for the frame #2
```

> Things that do not exist in any form yet: Eval on a specific stack frame or changing the selectedFrame.

There is an ancient tracking bug for this issue here: https://github.com/nodejs/node-inspect/issues/47

**Comment by alexp1917** [comment]
I agree! I learned about the merits of line debugging during my time programming in java, for work, long after I got started programming in node. where to begin?

**Comment by wesgarland** [comment]
I'm not really sure where to begin implementing this feature, other than to look at the implementation of the bt commands that I posted above, and taking a look at the v8 inspector API.....the key I don't know is actually how to evaluate code in an different stack frame.

I think the commands to implement in niim would be `exec(X, cmd)` and `repl X`, where X is the stack frame number. 

nodejs#47 bug suggests that `currentFrame X` would also be a good way to implement this feature....and I agree.  So for repl and eval, if unspecified, X would be the current frame number, usually 0.  Then we add the commands up and down to increment and decrement the current frame, like in gdb.

The chrome dev tools implement evaluateOnCallFrame so this is definitely possible, see https://chromedevtools.github.io/devtools-protocol/tot/Debugger/#method-evaluateOnCallFrame

And...wow!  lib/internal/inspect_repl.js  already uses that call to implement evalInCurrentContext (line 522) ... looks like this might not be too hard.
